sapi/fpm/fpm/fpm_main.c

```diff
@@ -1836,25 +1841,50 @@ consult the installation file that came with this distribution, or visit \n\
 
 	/* library is already initialized, now init our request */
 	request = fpm_init_request(fcgi_fd);
+	int fpm_bootstrapped = CGIG(fpm_bootstrap) && CGIG(fpm_bootstrap)[0];
 
 	zend_first_try {
-		while (EXPECTED(fcgi_accept_request(request) >= 0)) {
+		while (1) {
+			// moving this before init_request_info will break with dtrace
+			// support in php_request_startup(), can we remove?
+			if (UNEXPECTED(fpm_bootstrapped && php_request_startup() == FAILURE)) {
```

Comment:
NIT: you should probably move this to the condition below (no need to check `fpm_bootstrapped` twice)